### PR TITLE
Delete the build scripts that were copied from manylinux project

### PR DESCRIPTION
**Pull request by snnn**
**Description**: 

1. Delete the build scripts that were copied from manylinux project. Use "git checkout" instead.
2. Update manylinux version to get python 3.11. Related issue: #12343
3. Change the cuda version of linux gpu build job of nuget packaging pipeline from cuda 11.4 to cuda 11.6 to match the TRT job within the same pipeline.. (A lot other places need be updated as well, but I'd prefer to put them in another PR)
4. Make dockerfile names static. For example, replace `tools/ci_build/github/linux/docker/$(DockerFile)` to `tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_cpu` . The former one relies on a runtime variable `$(DockerFile)`, Template Parameters are expanded early in processing a [pipeline run](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/runs?view=azure-devops) when most variables are not available. It like C++ macros vs variables. 

**Motivation and Context**
- Why is this change required? What problem does it solve?

Would be easier to maintain. 

- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: tools/ci_build/github/azure-pipelines/templates/rocm.yml [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    clean: true
+    submodules: recursive
+
+  - template: set-python-manylinux-variables-step.yml
+  - template: get-docker-image-steps.yml
+    parameters:
+      Dockerfile: tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_rocm${{ parameters.RocmVersion }}
+      Context: tools/ci_build/github/linux/docker
+      DockerBuildArgs: >-
+        --build-arg TORCH_VERSION=1.11.0
```
Comment (mszhanyi) [inline tools/ci_build/github/azure-pipelines/templates/rocm.yml:34]: PyTorch latest version is 1.12

File: tools/ci_build/github/azure-pipelines/linux-ci-pipeline.yml [hunk 2]
```diff
@@ -1,3 +1,11 @@
+resources:
```
Comment (mszhanyi) [inline tools/ci_build/github/azure-pipelines/linux-ci-pipeline.yml:1]: so many duplicated resource snippets, could we put them in one template

File: tools/ci_build/github/azure-pipelines/templates/rocm.yml [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+          -e NIGHTLY_BUILD \
+          -e BUILD_BUILDNUMBER \
+          --user onnxruntimedev \
+          onnxruntimetrainingrocmbuild-rocm${{ parameters.RocmVersion }} \
+            /onnxruntime_src/tools/ci_build/build.py \
+              --config Release \
+              --use_rocm \
+              --rocm_version=${{ parameters.RocmVersion }} \
+              --rocm_home=/opt/rocm \
+              --nccl_home=/opt/rocm \
```
Comment (mszhanyi) [inline tools/ci_build/github/azure-pipelines/templates/rocm.yml:65]: is the --nccl_home useful here?

File: tools/ci_build/github/azure-pipelines/templates/rocm.yml [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    displayName: 'Find video and render gid to be mapped into container'
+
+  - task: CopyFiles@2
+    displayName: 'Copy Python Wheel to: $(Build.ArtifactStagingDirectory)'
+    inputs:
+      SourceFolder: '$(Build.BinariesDirectory)'
+      Contents: 'Release/dist/*.whl'
+      TargetFolder: '$(Build.ArtifactStagingDirectory)'
+
+  - task: CmdLine@2
```
Comment (mszhanyi) [inline tools/ci_build/github/azure-pipelines/templates/rocm.yml:100]: The task looks unnecessary. Please check the mail Xavier Dupre replied on May 18th

File: tools/ci_build/github/azure-pipelines/templates/rocm.yml (hunk 1 above)
Comment (snnn) [reply tools/ci_build/github/azure-pipelines/templates/rocm.yml:34]: The rocm team decides which pytorch version to use. We will not change it.

File: tools/ci_build/github/azure-pipelines/linux-ci-pipeline.yml (hunk 2 above)
Comment (snnn) [reply tools/ci_build/github/azure-pipelines/linux-ci-pipeline.yml:1]: No. Azure devops doesn't have this feature for resources.

File: tools/ci_build/github/azure-pipelines/templates/rocm.yml (hunk 3 above)
Comment (snnn) [reply tools/ci_build/github/azure-pipelines/templates/rocm.yml:65]: I don't know. I just keep the original code.

File: tools/ci_build/github/azure-pipelines/templates/rocm.yml (hunk 4 above)
Comment (snnn) [reply tools/ci_build/github/azure-pipelines/templates/rocm.yml:100]: I would prefer to keep it unchanged. Even if the task is unnecessary, should the owner of this pipeline to delete it.

File: tools/ci_build/github/azure-pipelines/templates/rocm.yml [hunk 5]
```diff
@@ -0,0 +1,149 @@
+parameters:
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/templates/rocm.yml:1]: Contents of this file are copied from https://github.com/microsoft/onnxruntime/blob/master/tools/ci_build/github/azure-pipelines/orttraining-py-packaging-pipeline-rocm.yml

File: tools/ci_build/github/azure-pipelines/templates/rocm.yml (hunk 1 above)
Comment (snnn) [reply tools/ci_build/github/azure-pipelines/templates/rocm.yml:30]: This file is for ROCM EP. Each EP's owners determine which CUDA version/ROCM version/... they want to use. Usually we don't make changes on it.

Comment (snnn) [reply tools/ci_build/github/azure-pipelines/templates/rocm.yml:30]: They do not always want to use the latest.

File: cgmanifests/generated/cgmanifest.json [hunk 6]
```diff
@@ -28,19 +28,8 @@
         "Type": "other",
```
Comment (snnn) [inline cgmanifests/generated/cgmanifest.json:28]: This file is autogenerated.

**Review by jchen351 — APPROVED** [review]

**Pull request merged**